cmd/root.go

```diff
@@ -56,30 +60,33 @@ func init() {
 }
 
 func initConfig() {
-	configOverride, _ := RootCmd.Flags().GetString("config")
-	if DefaultConfig == "" {
-		raw, err := os.ReadFile("conf/importer-example.yml")
-		if err == nil {
-			DefaultConfig = string(raw)
-		} else {
-			DefaultConfig = ""
+	once.Do(func() {
+		configOverride, _ := RootCmd.Flags().GetString("config")
+		if DefaultConfig == "" {
+			raw, err := os.ReadFile("config/importer-example.yml")
+			if err == nil {
+				DefaultConfig = string(raw)
+			} else {
+				DefaultConfig = ""
+			}
 		}
-	}
-	config.InitConfig(configOverride, DefaultConfig)
+		config.InitConfig(configOverride, DefaultConfig)
 
-	setupGrafanaClient()
-	log.Debug("Creating output locations")
-	//Output Renderer
-	TableObj = table.NewWriter()
-	TableObj.SetOutputMirror(os.Stdout)
-	TableObj.SetStyle(table.StyleLight)
+		grafanaSvc = service.NewApiService()
+		//Output Renderer
+		TableObj = table.NewWriter()
+		TableObj.SetOutputMirror(os.Stdout)
+		TableObj.SetStyle(table.StyleLight)
 
-	if config.Config().IsDebug() {
-		log.SetLevel(log.DebugLevel)
-	}
-}
-
-func setupGrafanaClient() {
-	grafanaSvc = service.NewApiService()
+		if config.Config().IsDebug() {
+			log.SetLevel(log.DebugLevel)
+		}
+		//Validate current configuration
+		valid := config.Config().GetDefaultGrafanaConfig().Validate()
+		if !valid {
```

Comment:
This feels like a bad idea to me. Is there any reason why we shouldn't force them to fix the config instead of potentially introducing errors / undefined behavior in other places?

Reply:
Okay, fair enough.  Updated.